### PR TITLE
User list refactor

### PR DESCRIPTION
**Pull request by frankyma**
Copied from project list refactor.  Skipped the first/last name filters as Carson said we don't need them.  Was in the process of making `applyListFilters` to abstract the users but will keep the code in, maybe used later.

### CODE REVIEW

File: src/core/array-helpers.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+/**
```
Comment (CarsonF) [inline src/core/array-helpers.ts:1]: Can you put these in `common/util`?

File: src/core/database/query.helpers.ts [hunk 2]
```diff
@@ -1102,3 +1103,19 @@ export function getPropList(query: Query, ...aliases: string[]) {
     ])
     .with(['collect(prop) as propList', 'node', ...aliases]);
 }
+
+//List filtering after refactor
+export function applyListFilters<
```
Comment (CarsonF) [inline src/core/database/query.helpers.ts:1108]: Let's not do this. Each filter property can have different handling so handling as a complete object will not work.

File: src/core/array-helpers.ts (hunk 1 above)
Comment (frankyma) [reply src/core/array-helpers.ts]: Moved

File: src/core/database/query.helpers.ts (hunk 2 above)
Comment (frankyma) [reply src/core/database/query.helpers.ts]: Removed 😢

**Comment by zkhin** [comment]
@CarsonF Do we still need `displayFirstName` and `displayLastName` in the user list filter?

**Comment by CarsonF** [comment]
No

**Review by CarsonF — APPROVED** [review]
Thanks @frankyma & @zkhin!

**Review by CarsonF — DISMISSED** [review]

**Comment by CarsonF** [comment]
@zkhin Can you update user tests to account for the fact that you can't filter now? Maybe the tests can just be removed? I didn't look hard enough to confirm whether they are still applying value or not.

**Comment by zkhin** [comment]
> @zkhin Can you update user tests to account for the fact that you can't filter now? Maybe the tests can just be removed? I didn't look hard enough to confirm whether they are still applying value or not.

Okay, I will

**Review by CarsonF — APPROVED** [review]
💪

**Pull request merged**